### PR TITLE
Reduced Cost Spoke & Fixer

### PR DESCRIPTION
**Pull request by llueg**
Implementation of reduced cost heuristics & bound tightening.
- reduced cost spoke: solves relaxed subproblems and computes reduced costs
- reduced costs extension: performs bound tightening based on RC from RC spoke. Performs heuristic fixing of variables with large RCs, given some user-defined fix fraction (i.e. fixes 70% of variables with largest abs. RC).
- ph-tracking for reduced costs
- configurations for RC spoke and RC fixer
- modified some cylinder files to include RC options
- added one test run for sslp with RC spoke in run_all

### CODE REVIEW

**Review by DLWoodruff — COMMENTED** [review]
There should be some (short, perhaps) documentation in doc/src/spokes.rst

File: mpisppy/extensions/phtracker.py
Comment (DLWoodruff) [inline mpisppy/extensions/phtracker.py:1]: Thanks for all the bug fixes.

**Review by DLWoodruff — APPROVED** [review]
Looks great. Thanks for this addition.

File: mpisppy/utils/cfg_vanilla.py [hunk 1]
```diff
@@ -196,6 +201,26 @@ def add_cross_scenario_cuts(hub_dict,
...
+            "use_rc_fixer": cfg.rc_fixer,
+            "zero_rc_tol": cfg.rc_zero_tol,
+            "fix_fraction_target_iter0": cfg.rc_fix_fraction_iter0,
+            "fix_fraction_target_iterK": cfg.rc_fix_fraction_iterk,
+            "use_rc_bt": cfg.rc_bound_tightening,
+            "rc_bound_tol": cfg.rc_bound_tol,
+        }
+
+
+#     return hub_dict
```
Comment (bknueven) [inline mpisppy/utils/cfg_vanilla.py:222]:
```suggestion

     return hub_dict
```

**Review by bknueven — APPROVED** [review]

File: mpisppy/utils/cfg_vanilla.py [hunk 2]
```diff
@@ -196,6 +201,25 @@ def add_cross_scenario_cuts(hub_dict,
...
+            "debug": cfg.rc_debug,
+            "use_rc_fixer": cfg.rc_fixer,
+            "zero_rc_tol": cfg.rc_zero_tol,
+            "fix_fraction_target_iter0": cfg.rc_fix_fraction_iter0,
+            "fix_fraction_target_iterK": cfg.rc_fix_fraction_iterk,
+            "use_rc_bt": cfg.rc_bound_tightening,
+            "rc_bound_tol": cfg.rc_bound_tol,
+        }
+
+     return hub_dict
```
Comment (bknueven) [inline mpisppy/utils/cfg_vanilla.py:221]:
```suggestion
        }

    return hub_dict
```

**Pull request merged**